build_tools/cmake/iree_copts.cmake

```diff
@@ -129,6 +129,10 @@ set(LLVM_TARGETS_TO_BUILD "WebAssembly;X86" CACHE STRING "" FORCE)
 set(LLVM_ENABLE_PROJECTS "mlir" CACHE STRING "" FORCE)
 set(LLVM_ENABLE_BINDINGS OFF CACHE BOOL "" FORCE)
 
+if(IREE_USE_LINKER)
+  set(LLVM_USE_LINKER ${IREE_USE_LINKER} CACHE STRING "" FORCE)
+endif()
+
```

Comment:
Maybe a check similar to https://github.com/llvm/llvm-project/blob/760f45eacadbabf9634fb81d7ccaa16c269cf19e/llvm/cmake/modules/HandleLLVMOptions.cmake#L260-L265 should be introduced here?